Kudu.Services/HttpRequestExtensions.cs

```diff
@@ -9,13 +10,16 @@ public static class HttpRequestExtensions
         public static Stream GetInputStream(this HttpRequestBase request)
         {
             var contentEncoding = request.Headers["Content-Encoding"];
-
             if (contentEncoding != null && contentEncoding.Contains("gzip"))
             {
-                return new GZipStream(request.GetBufferlessInputStream(), CompressionMode.Decompress);
+                // https://github.com/mono/mono/pull/1914
+                // GetBufferlessInputStream will come in next release (current 4.2.1)
+                return new GZipStream(request.InputStream, CompressionMode.Decompress);
```

Comment:
But doesn't that mean that we lose an optimization on Windows? Should this be Linux conditional?